sdk/core/core-https/src/nodeHttpsClient.ts

```diff
@@ -198,9 +198,17 @@ class NodeHttpsClient implements HttpsClient {
     const proxySettings = request.proxySettings;
     if (proxySettings) {
       if (!this.proxyAgent) {
+        let parsedUrl: URL;
+        try {
+          parsedUrl = new URL(proxySettings.host);
+        } catch (_error) {
+          throw new RangeError(`Expecting a valid host string in proxy settings, but found "${proxySettings.host}".`);
```

Comment:
Interesting, I always thought `RangeError` was meant to be used if either a number fell outside of an allowable range, or if a string wasn't included in some known list of acceptable values (e.g. enums). Do we use this for pattern matching errors as well?

Reply:
I was thinking `RangeError` can be used for invalid parameters (as they are not in the set of allowed values), but I can changed it to `Error`.  I checked guideline on parameter validation, but it doesn't say which type of errors we should throw.

Reply:
Found this in guideline so I will use `Error` instead
![image](https://user-images.githubusercontent.com/7583839/108922219-94d9d400-75eb-11eb-8f8d-21e34d3b6bcb.png)